internal/app/secretless/tcp_proxy_service.go

```diff
@@ -0,0 +1,147 @@
+package secretless
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"reflect"
+
+	validation "github.com/go-ozzo/ozzo-validation"
+
+	"github.com/cyberark/secretless-broker/internal"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/tcp"
+)
+
+// Zeroizes the values of the fetched credentials. We don't want to
+// rely on garbage collection for this (it might be slow and/or only free them) so
+// we manually clear
+func zeroizeCredentials(backendCredentials map[string][]byte) {
```

Comment:
how does this work if it passes the argument in by value?

Reply:
a map is a pointer, see https://dave.cheney.net/2017/04/30/if-a-map-isnt-a-reference-variable-what-is-it

Reply:
TIL. Thanks for the link @doodlesbykumbi!